controllers/service.go

```diff
@@ -21,77 +21,64 @@ func getSTSHeadLessSvcName(aeroCluster *asdbv1.AerospikeCluster) string {
 	return aeroCluster.Name
 }
 
-func (r *SingleClusterReconciler) createSTSHeadlessSvc() error {
-	r.Log.Info("Create headless service for statefulSet")
-
-	ls := utils.LabelsForAerospikeCluster(r.aeroCluster.Name)
+func (r *SingleClusterReconciler) createOrUpdateSTSHeadlessSvc() error {
+	r.Log.Info("Create or Update headless service for statefulSet")
 
 	serviceName := getSTSHeadLessSvcName(r.aeroCluster)
-	service := &corev1.Service{}
 
-	err := r.Client.Get(
-		context.TODO(), types.NamespacedName{
-			Name: serviceName, Namespace: r.aeroCluster.Namespace,
-		}, service,
-	)
-	if err != nil {
-		if errors.IsNotFound(err) {
-			service = &corev1.Service{
-				ObjectMeta: metav1.ObjectMeta{
-					// Headless service has the same name as AerospikeCluster
-					Name:      serviceName,
-					Namespace: r.aeroCluster.Namespace,
-					// deprecation in 1.10, supported until at least 1.13,  breaks peer-finder/kube-dns if not used
-					Annotations: map[string]string{
-						"service.alpha.kubernetes.io/tolerate-unready-endpoints": "true",
-					},
-					Labels: ls,
-				},
-				Spec: corev1.ServiceSpec{
-					// deprecates service.alpha.kubernetes.io/tolerate-unready-endpoints as of 1.
-					// 10? see: kubernetes/kubernetes#49239 Fixed in 1.11 as of #63742
-					PublishNotReadyAddresses: true,
-					ClusterIP:                "None",
-					Selector:                 ls,
-				},
-			}
-
-			r.appendServicePorts(service)
+	err := r.updateServicePorts(serviceName, r.aeroCluster.Namespace)
```

Comment:
This flow doesn't seem much readable. Instead we can get the service first here and based on that can go with create or update flow similar to old code